packages/blockchain/polygon/src/lib/services/polygon.virtualAccount.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  AccountService,
+  ApiServices,
+  Currency,
+  TransferEthKMS,
+  TransferEth,
+  VirtualCurrencyService,
+  WithdrawalService,
+} from '@tatumio/api-client'
+import {
+  abstractBlockchainVirtualAccount,
+  PrivateKeyOrSignatureId,
+} from '@tatumio/shared-blockchain-abstract'
+import { evmBasedUtils, EvmBasedWeb3 } from '@tatumio/shared-blockchain-evm-based'
+import { Blockchain } from '@tatumio/shared-core'
+import BigNumber from 'bignumber.js'
+import { polygonTxService } from './polygon.tx'
+
+type TransferVirtualAccountPolygon = PrivateKeyOrSignatureId<TransferEth>
+type VirtualAccountResponse = { id?: string; txId?: string; completed?: boolean } | void
+
+const sendPolygonVirtualAccountTransaction = async (
+  body: TransferVirtualAccountPolygon,
+  web3: EvmBasedWeb3,
+): Promise<VirtualAccountResponse> => {
+  const txService = polygonTxService({ blockchain: Blockchain.POLYGON, web3 })
+  const { mnemonic, index, privateKey, gasLimit, gasPrice, nonce, ...withdrawal } = body
+  const { amount, address } = withdrawal
+  let fromPrivateKey: string
+  let txData: string
+
+  if (mnemonic && index !== undefined) {
+    fromPrivateKey = await evmBasedUtils.generatePrivateKeyFromMnemonic(Blockchain.POLYGON, mnemonic, index)
+  } else {
+    fromPrivateKey = privateKey as string
+  }
+
+  const account = await AccountService.getAccountByAccountId(body.senderAccountId)
+  const fee = {
+    gasLimit: gasLimit || '21000',
+    gasPrice: gasPrice || '20',
+  }
+
+  if (account.currency === Currency.MATIC) {
+    txData = await txService.native.prepare.transferSignedTransaction({
+      amount,
+      fromPrivateKey,
+      fee,
+      nonce,
+      to: address,
+    })
+  } else {
+    fee.gasLimit = '100000'
+    const vc = await VirtualCurrencyService.getCurrency(account.currency)
```

Comment:
you dont cover here cases for predefined URC20 tokens like USDT_MATIC and others - check BSC wehre Marina has done that already